src/main/scala/is/hail/annotations/UnsafeRow.scala

```diff
@@ -1,20 +1,56 @@
 package is.hail.annotations
 
-import java.io.{DataInputStream, DataOutputStream}
+import java.io.{DataInputStream, DataOutputStream, ObjectInputStream, ObjectOutputStream}
 
 import com.esotericsoftware.kryo.io.{Input, Output}
 import com.esotericsoftware.kryo.{Kryo, KryoSerializable}
 import is.hail.expr._
 import is.hail.utils.Interval
 import is.hail.variant.{AltAllele, GenericGenotype, GenomeReference, Locus, Variant}
+import org.apache.spark.SparkContext
 import org.apache.spark.broadcast.Broadcast
 import org.apache.spark.sql.Row
 
-class UnsafeIndexedSeqAnnotation(
+object UnsafeIndexedSeq {
+  def apply(sc: SparkContext, region: MemoryBuffer, t: TArray, aoff: Long, length: Int): UnsafeIndexedSeq = {
+    new UnsafeIndexedSeq(region, BroadcastTypeTree(sc, t), t.elementByteSize, aoff, aoff + t.elementsOffset(length), length)
+  }
+
+  def apply(ttBc: Broadcast[TypeTree], region: MemoryBuffer, aoff: Long): UnsafeIndexedSeq = {
```

Comment:
I think we should rip out the typetree stuff. It's unnecessary complexity to hack performance in a way that won't matter when the unsafe system is complete